tests/specs/controllers/CheckUploadedDocuments.test.js

```diff
@@ -0,0 +1,376 @@
+            const expectedArg = {
+                where: {
+                    application_id: 12345,
+                },
+            };
+            expect(findApplicationPaymentDetails.calledWith(expectedArg)).to.be
+                .true;
+        });
+
+        it('should update the payment details if an entry exists ', () => {
+            // when
+            const findApplicationPaymentDetails = sandbox.stub(
+                ApplicationPaymentDetails,
+                'find'
+            );
+            findApplicationPaymentDetails.resolves(true);
+
+            const updatePaymentAmount = sandbox.spy(
+                CheckUploadedDocumentsController,
+                '_updatePaymentAmountInDB'
+            );
+
+            CheckUploadedDocumentsController.addDocsToDBHandler(
+                reqStub,
+                resStub
+            );
+
+            // then
+            assertWhenPromisesResolved(
+                () => expect(updatePaymentAmount.calledOnce).to.be.true
+            );
+        });
+
+        it('should create new payment details if an entry does NOT exists ', () => {
+            // when
+            const findApplicationPaymentDetails = sandbox.stub(
+                ApplicationPaymentDetails,
+                'find'
+            );
+            findApplicationPaymentDetails.resolves(false);
+
+            const createPaymentDetails = sandbox.spy(
+                CheckUploadedDocumentsController,
+                '_createPaymentDetailsInDB'
+            );
+
+            CheckUploadedDocumentsController.addDocsToDBHandler(
+                reqStub,
+                resStub
+            );
+
+            // then
+            assertWhenPromisesResolved(
+                () => expect(createPaymentDetails.calledOnce).to.be.true
+            );
+        });
+
+    });
+
+    describe('_checkPaymentDetailsExistsInDB', () => {
+        it('redirects to payment page after document count and payment details checks', () => {
+            // when
+            const createUploadedDocumentsUrls = sandbox.stub(
+                UploadedDocumentUrls,
+                'create'
+            );
+            const findUserDocumentCount = sandbox.stub(
+                UserDocumentCount,
+                'find'
+            );
+            const updateUserDocumentCount = sandbox.stub(
+                UserDocumentCount,
+                'update'
+            );
+            const findApplicationPaymentDetails = sandbox.stub(
+                ApplicationPaymentDetails,
+                'find'
+            );
+            const updateApplicationPaymentDetails = sandbox.stub(
+                ApplicationPaymentDetails,
+                'update'
+            );
+            createUploadedDocumentsUrls.resolves();
+            findUserDocumentCount.resolves(true);
+            updateUserDocumentCount.resolves();
+            findApplicationPaymentDetails.resolves(true);
+            updateApplicationPaymentDetails.resolves();
```

Comment:
double check this, but I'm pretty sure you don't need to declare a spy (and assign it) to be able to mock out what it resolves.

I _think_ you can do something like 
```
sandbox.stub(UploadedDocumentUrls, 'create').resolves()
```
this is better since it doesn't look like you are using these spies anyway, just stubbing them.

Reply:
also, if you don't know already, it's worth reading up on dependency injection / inversion of control and how it makes testing easier in OOP. It's probably not right for this project now, since it could be a big refactor which we don't have time for

Reply:
Ah thanks for that, I've read about DI before but never really looked into how it would help tests. Unfortunately, you're right about it being a refactor we don't have time for but definitely something to keep in mind for a greenfield project.